cmake/detect-intrinsics.cmake

```diff
@@ -2,35 +2,34 @@
 # Licensed under the Zlib license, see LICENSE.md for details
 
 macro(check_acle_compiler_flag)
-    if(MSVC)
-        # Both ARM and ARM64-targeting msvc support intrinsics, but
-        # ARM msvc is missing some intrinsics introduced with ARMv8, e.g. crc32
-        if(MSVC_C_ARCHITECTURE_ID STREQUAL "ARM64")
-            set(HAVE_ACLE_FLAG TRUE)
-        endif()
-    else()
-        if(NOT NATIVEFLAG)
-            if(CMAKE_C_COMPILER_ID MATCHES "GNU" OR CMAKE_C_COMPILER_ID MATCHES "Clang")
+    if(NOT NATIVEFLAG)
+        if(CMAKE_C_COMPILER_ID MATCHES "GNU" OR CMAKE_C_COMPILER_ID MATCHES "Clang")
+            check_c_compiler_flag("-march=armv8-a+crc" HAVE_MARCH_ARMV8_CRC)
+            if(HAVE_MARCH_ARMV8_CRC)
                 set(ACLEFLAG "-march=armv8-a+crc" CACHE INTERNAL "Compiler option to enable ACLE support")
+            else()
+                check_c_compiler_flag("-march=armv8-a+crc+simd" HAVE_MARCH_ARMV8_CRC_SIMD)
+                if(HAVE_MARCH_ARMV8_CRC_SIMD)
+                    set(ACLEFLAG "-march=armv8-a+crc+simd" CACHE INTERNAL "Compiler option to enable ACLE support")
+                endif()
```

Comment:
**question:** 
Would it make sense to instead test for `-march=armv8-a+crc+simd` before falling back to `-march=armv8-a+crc` instead of preferring to not use the +simd option?
I am not intimately familiar with what the +simd option gives us in practice, but it seems logical to instead use the one without it as the fallback here.

Reply:
I'm not too familiar with cmake, the changes should be functionally the same as the ones to the configure script.

Reply:
True enough. I don't know why it is like that there either. 🤷‍♂️

Reply:
Should I mark this as resolved or would you like me to switch the order of the flags?

Reply:
I am not familiar enough with this, so I was hoping someone who is would add their thoughts on the matter.

The PR is also waiting for a review by someone else, I try not to merge things until someone else in addition to me has reviewed it, and especially when I am not feeling confident in whether it is correct or not.

Reply:
I'm not sure about it either. It does look like a simplification to some degree.

Reply:
Testing `crc+simd` first would mean we can't disable NEON support... Some compilers require both extensions to be enabled to be able to target either one because NEON uses different calling convention.

Reply:
@mtl1979 Thanks, that makes sense. But do we take this into account when do the runtime tests as well?

And would it be possible for you to do a review of this PR? I'd like at least one other approval before I merge it. To me it looks ok, so I will merge it eventually even without any other reviews.

Reply:
@Dead2 Runtime check isn't really affected as the issue is just with the command-line parameters... Some compilers don't enable FPU, which means NEON can't be used. By default FPU should be enabled when available.